pkg/apis/sources/v1alpha1/interfaces.go

```diff
@@ -58,6 +59,31 @@ func IsMultiTenant(src EventSource) bool {
 	return ok && mt.IsMultiTenant()
 }
 
+// serviceAccountProvider is implemented by source types which are able to
+// influence the shape of the ServiceAccount used by their own receive adapter.
+type serviceAccountProvider interface {
+	WantsOwnServiceAccount() bool
+	ServiceAccountOptions() []func(*corev1.ServiceAccount)
+}
+
+// WantsOwnServiceAccount returns whether the given source instance should have
+// a dedicated ServiceAccount associated with its receive adapter.
+func WantsOwnServiceAccount(src EventSource) bool {
+	saProvider, ok := src.(serviceAccountProvider)
+	return ok && saProvider.WantsOwnServiceAccount()
+}
+
+// ServiceAccountOptions returns functional options for mutating the
+// ServiceAccount associated with a given source instance.
+func ServiceAccountOptions(src EventSource) []func(*corev1.ServiceAccount) {
+	saProvider, ok := src.(serviceAccountProvider)
+	if !ok {
+		return nil
```

Comment:
Since the return type is bool, shouldn't this be `false`?

Reply:
On this method the return type is `[]func(*corev1.ServiceAccount)`.
The other method `WantsOwnServiceAccount()` is the one which returns a boolean.

There are two methods because a source may want its own ServiceAccount but not necessarily need to mutate the shape of that ServiceAccount (just to be able to bind it to a custom RBAC role, for example).

Reply:
Derp, okay I'll make an appointment to see the optometrist again as my eyes wen towards line 71 instead of 78 for that.